src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -41,13 +40,19 @@ internal class CSharpType
         }
 
         public CSharpType(Type type, params CSharpType[] arguments) : this(type, false, arguments)
+        { }
+
+        public CSharpType(Type type, IReadOnlyList<CSharpType> arguments) : this(type, false, arguments)
+        { }
+
+        public CSharpType(Type type, bool isNullable, params CSharpType[] arguments) : this(type, isNullable, arguments as IReadOnlyList<CSharpType>)
```

Comment:
this ` as IReadOnlyList<CSharpType>` here is required because without it this ctor is actually calling itself and leads to a compilation error.

Reply:
We still need to keep the constructor on line 43?

Reply:
That `params` ctor is used in a lot of places, especially when there is only one generic argument, such as:
```
new CSharpType(typeof(IList<>), typeof(string))
```
if we remove this, we have to use it like this:
```
new CSharpType(typeof(IList<>), new CSharpType[] {typeof(string)})
```
which is cumbersome and inconvenient.

I do not like the solution of adding a ctor for the case we have one generic parameter, which is another confusion with the other ctor `CSharpType(Type type, Type? serializeAs)`